### ISSUE TITLE
Add a way to cleanup a feature

### ISSUE THREAD

**Issue by laeubi** [eclipse-pde/eclipse.pde#880]
It would be good to have a way in PDE (similar to the Manifest cleanup actions), that allows to cleanup a feature in a way where it removes everything that is not a "root" item.

e.g. consider I have a feature that includes bundles `A` + `B`, now there is a dependency chain from `A` > `X` > `B` then the cleanup action should remove `B` from the feature as it is already included. Similar things should work for included features.

Currently one has to do this tedious work manually (and carefully) if one wants to only mention the real required items.
FYI @akurtakov @merks @HannesWell

**Comment by HannesWell** [comment]
Yes that's useful and was already discussed as part of https://github.com/eclipse-pde/eclipse.pde/discussions/331#discussioncomment-3581163.
Locally I started to implement something back then, but I have not yet completed it.
In general this is a similar task and can share a lot of code.

But in this functionality can probably be a (probably NP) hard problem if multiple included features have partial overlap of (transitively) included plugins and multiple features can be removed, but minimizing the number of included features can be difficult (I haven't looked it up yet, but it is probably one of the standard optimization problems).
Nevertheless one can start with a simple solution that works for the obvious/clear cases. Most of the time, the problem instances are of good nature anyways.
